src/components/RequesterPaysModal.test.tsx

```diff
@@ -1,65 +1,86 @@
 import { screen } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
 import React from 'react';
-import { renderWithAppContexts as render } from 'src/testing/test-utils';
+import { asMockedFn, renderWithAppContexts as render } from 'src/testing/test-utils';
+import { useWorkspaces } from 'src/workspaces/common/state/useWorkspaces';
 import { WorkspaceWrapper } from 'src/workspaces/utils';
 
 import RequesterPaysModal from './RequesterPaysModal';
 
-jest.mock('src/workspaces/common/state/useWorkspaces', () => ({
-  ...jest.requireActual('src/workspaces/common/state/useWorkspaces'),
-  useWorkspaces: jest.fn().mockReturnValue({
-    loading: false,
-    workspaces: [
-      {
-        workspace: {
-          namespace: 'test-namespace',
-          name: 'workspace-1',
-          cloudPlatform: 'Gcp',
-          googleProject: 'test-project-1',
+type UseWorkspacesExports = typeof import('src/workspaces/common/state/useWorkspaces');
+jest.mock('src/workspaces/common/state/useWorkspaces', (): UseWorkspacesExports => {
+  return {
+    ...jest.requireActual<UseWorkspacesExports>('src/workspaces/common/state/useWorkspaces'),
+    useWorkspaces: jest.fn(),
+  };
+});
+
+describe('RequesterPaysModal', () => {
+  it('lists GCP workspaces that the user owns or is writer with canCompute', async () => {
+    // Arrange
+    asMockedFn(useWorkspaces).mockReturnValue({
```

Comment:
Moved this into the test methods so that we can verify the correct fields are passed when `useWorkspaces` is called.

Also changed some of the workspace names to make it more obvious what is being tested. And added coverage for writer with and without `canCompute`.